zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/UserTaskController.java

```diff
@@ -120,4 +130,34 @@ private static ProblemDetail mapRejectionToProblem(final BrokerRejection rejecti
         }
     };
   }
+
+  @PostMapping(
+      path = "/user-tasks/search",
+      produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_PROBLEM_JSON_VALUE},
+      consumes = MediaType.APPLICATION_JSON_VALUE)
+  public ResponseEntity<UserTaskSearchQueryResponse> searchUserTasks(
+      @RequestBody(required = false) final UserTaskSearchQueryRequest query) {
+    return SearchQueryRequestMapper.toUserTaskQuery(query == null? new UserTaskSearchQueryRequest() : query)
+        .fold(
+            (Function<? super UserTaskQuery, ? extends ResponseEntity<UserTaskSearchQueryResponse>>)
+                this::search,
+            RestErrorMapper::mapProblemToResponse);
+  }
+
+  private ResponseEntity<UserTaskSearchQueryResponse> search(final UserTaskQuery query) {
+    try {
+      final var tenantIds = TenantAttributeHolder.tenantIds();
+      final var result =
+          userTaskServices.withAuthentication((a) -> a.tenants(tenantIds)).search(query);
+      return ResponseEntity.ok(
+          SearchQueryResponseMapper.toUserTaskSearchQueryResponse(result).get());
+    } catch (final Throwable e) {
+      final var problemDetail =
+          RestErrorMapper.createProblemDetail(
+              HttpStatus.BAD_REQUEST, e.getMessage(), "Failed to execute UserTask Search Query");
```

Comment:
❓ Is this really going to be a 400 response each time? At this point, can you still differentiate based on the error what went wrong? Will it rather be non-retriable errors, so 500 responses?

Reply:
This is true, I believe 400 can be filtered by Validation Exception, and other ones can be handled by 500. I applied the suggestion over here https://github.com/camunda/camunda/pull/19625/commits/960e60406b7e0235cc42f61dfb67ee2f2fcc2452